packages/reporters/build-metrics/src/BuildMetricsReporter.js

```diff
@@ -4,12 +4,44 @@ import path from 'path';
 import {Reporter} from '@parcel/plugin';
 import {generateBundleReport} from '@parcel/utils';
 
+type TimingValue = {|
+  timings: {[key: string]: number, ...},
+  lastPhase: string,
+|};
+
+let timingsMap = new Map();
+const getValue = (instanceId: string): TimingValue => {
+  if (!timingsMap.has(instanceId)) {
+    timingsMap.set(instanceId, {
+      timings: {},
+      lastPhase: 'resolving',
+    });
+  }
+
+  // $FlowFixMe
+  return timingsMap.get(instanceId);
+};
+
 export default new Reporter({
   async report({event, options}) {
-    if (event.type === 'buildSuccess') {
+    if (event.type === 'buildProgress') {
+      let value = getValue(options.instanceId);
+
+      value.timings[event.phase] = Date.now();
+      if (value.lastPhase !== event.phase) {
+        value.timings[value.lastPhase] =
+          Date.now() - value.timings[value.lastPhase];
+      }
+      value.lastPhase = event.phase;
+    } else if (event.type === 'buildSuccess') {
+      let value = getValue(options.instanceId);
+
+      value.timings[value.lastPhase] =
+        Date.now() - value.timings[value.lastPhase];
       let metricsFilePath = path.join(process.cwd(), 'parcel-metrics.json');
```

Comment:
Not entirely sure what path I'd have to use to save these all properly...

Reply:
The bundle analyzer also uses the project root
https://github.com/parcel-bundler/parcel/blob/2dc32a8b3f2113e01322f67bc5bc370bcd7be56d/packages/reporters/bundle-analyzer/src/BundleAnalyzerReporter.js#L28

Reply:
I guess I'd need to add `yarn.lock` to the benchmarks in that case, but sure projectRoot seems fine